src/SAMLTrace.js

```diff
@@ -402,72 +400,105 @@ SAMLTrace.RequestItem.prototype = {
   
   'showSummary' : function(target) {
     var doc = target.ownerDocument;
-    var samlSummary = "";
+    var samlSummary = '<div id="summary"><table>';
     var parser  = new DOMParser();
     var xmldoc  = parser.parseFromString(this.request.saml,"text/xml");
 
     /* Check for AuthnRequest */
     var AuthnRequest = xmldoc.getElementsByTagNameNS('*','AuthnRequest');
-    if (AuthnRequest.length>0) {
-        samlSummary += 'AuthnRequest: \n';
-        if (AuthnRequest[0].attributes['Destination'])                 { samlSummary += this.summaryAdd(AuthnRequest[0].attributes['Destination'].value                , 'Destination'                ); }
-        if (AuthnRequest[0].attributes['ForceAuthn'])                  { samlSummary += this.summaryAdd(AuthnRequest[0].attributes['ForceAuthn'].value                 , 'ForceAuthn'                 ); }
-        if (AuthnRequest[0].attributes['AssertionConsumerServiceURL']) { samlSummary += this.summaryAdd(AuthnRequest[0].attributes['AssertionConsumerServiceURL'].value, 'AssertionConsumerServiceURL'); }
+    if (AuthnRequest.length>0) { // We found AuthnRequest!
+        samlSummary += '<tr> <th colspan=2>AuthnRequest</th> </tr>';
+        for (let ArAtt of AuthnRequest[0].attributes) {
+            if (!ArAtt.name.startsWith('xmlns')) samlSummary += `<tr><td class="hljs-attribute">${ArAtt.name}</td><td> ${ArAtt.value}</td></tr>`;
+        }
+        /* Check for Issuer within AuthnRequest */
+        var Issuer = xmldoc.getElementsByTagNameNS('*','Issuer');
+        if (Issuer.length>0) samlSummary += `<tr><td class="hljs-attribute">Issuer</td><td> ${Issuer[0].textContent}</td></tr>`;
+    }
+
+
+    /*
+    Check for SAML2.0 Response
+    */
+    var SamlResponse = xmldoc.getElementsByTagNameNS('urn:oasis:names:tc:SAML:2.0:protocol','Response');
```

Comment:
I have a use case where a SAML2-token is transferred via WS-Fed. I'm not sure if this combination (WS-Fed + SAML2-Token instead of WS-Fed + SAML1.1-token) is very common, but anyway.

The assertion contained in that token won't be picked up by these lines nor by those at line 461.

If you changed line 423 from
`var SamlResponse = xmldoc.getElementsByTagNameNS('urn:oasis:names:tc:SAML:2.0:protocol','Response');`
to
`var SamlResponse = xmldoc.getElementsByTagNameNS('urn:oasis:names:tc:SAML:2.0:assertion','Assertion');`
it would just look for the assertion and not for the protocol-related nodes. This way it would just look for a SAML2-token. Regardless if it's transferred via SAML-protocol or WS-Fed! 🙂

I think this'd be a quite handy benefit 🙂

Reply:
Interesting. For sure my test cases are not broad enough :)

Anyway then I would rather try to separate parsing `SAML2.0 Response` from `SAML2.0 Assertion` because in other way we would loose attributes like `Issuer` or `Destination` if we go straight to `Assertion` as you suggested. 
Writing this I had quiet assumption that SAML2 tokens can be transferred only in SAML2 response. I should have some time this weekend to take a closer look on this. **Thank you very much for your suggestion**.

PS. it would be very useful if you could attach here .xml with yours use case. Of course "anonymize" everything possible in file :)

Reply:
Convertion from SAML2 to WS-Fed + SAML1.1 is not uncommon.. I pretty much rely on it!!

Reply:
@tvdijen that's true but I think (at least it is what I understood 🤔) that @khlr is writing about SAML2 inside FS-Fed (this case also can be not uncommon but I don't have any sample to do test against anyway 🙂)

Reply:
I have encountered the following combinations so far:
- SAML-protocol + SAML2.0-token
- WS-Fed + SAML1.1-token
- WS-Fed + SAML2.0-token (which I wasn't aware of yet 😁)

I'll try to generate and anonymize some examples. But it may take a little while again 😁➡🐌